Source/Core/AudioCommon/Mixer.h

```diff
@@ -54,16 +55,102 @@ class Mixer final
   static constexpr u64 FIXED_SAMPLE_RATE_DIVIDEND = 54000000 * 2;
 
 private:
-  static constexpr u32 MAX_SAMPLES = 1024 * 4;  // 128 ms
-  static constexpr u32 INDEX_MASK = MAX_SAMPLES * 2 - 1;
-  static constexpr int MAX_FREQ_SHIFT = 200;  // Per 32000 Hz
-  static constexpr float CONTROL_FACTOR = 0.2f;
-  static constexpr u32 CONTROL_AVG = 32;  // In freq_shift per FIFO size offset
-
   const unsigned int SURROUND_CHANNELS = 6;
 
   class MixerFifo final
   {
+    static constexpr size_t GRANULE_QUEUE_SIZE = 20;
+
+    /**
+     * import numpy as np
+     * import scipy.signal as signal
+     * window = np.convolve(np.ones(128), signal.windows.dpss(128 + 1, 3))
+     * window /= (window[:len(window) // 2] + window[len(window) // 2:]).max()
+     * window = np.round(window * (2 ** 16))
+     * elements = ", ".join([f"{int(x)}" for x in window])
+     * print(f'static constexpr std::array<int, {len(window)}> GRANULE_WINDOW = {{ {elements} }};')
+     */
+    static constexpr std::array<int, 256> GRANULE_WINDOW = {
+        1,     4,     7,     12,    19,    28,    39,    54,    72,    94,    122,   155,   194,
+        240,   295,   359,   433,   518,   615,   726,   852,   993,   1151,  1328,  1524,  1742,
+        1982,  2246,  2534,  2849,  3191,  3562,  3963,  4395,  4858,  5354,  5884,  6448,  7047,
+        7681,  8351,  9057,  9799,  10576, 11389, 12237, 13120, 14037, 14987, 15968, 16981, 18023,
+        19092, 20189, 21309, 22452, 23616, 24798, 25996, 27208, 28432, 29664, 30903, 32146, 33390,
+        34633, 35872, 37104, 38328, 39540, 40738, 41920, 43084, 44227, 45347, 46444, 47513, 48555,
+        49568, 50549, 51499, 52416, 53299, 54147, 54960, 55737, 56479, 57185, 57855, 58489, 59088,
+        59652, 60182, 60678, 61141, 61573, 61974, 62345, 62687, 63002, 63290, 63554, 63794, 64012,
+        64208, 64385, 64543, 64684, 64810, 64921, 65018, 65103, 65177, 65241, 65296, 65342, 65381,
+        65414, 65442, 65464, 65482, 65497, 65508, 65517, 65524, 65529, 65532, 65535, 65535, 65532,
+        65529, 65524, 65517, 65508, 65497, 65482, 65464, 65442, 65414, 65381, 65342, 65296, 65241,
+        65177, 65103, 65018, 64921, 64810, 64684, 64543, 64385, 64208, 64012, 63794, 63554, 63290,
+        63002, 62687, 62345, 61974, 61573, 61141, 60678, 60182, 59652, 59088, 58489, 57855, 57185,
+        56479, 55737, 54960, 54147, 53299, 52416, 51499, 50549, 49568, 48555, 47513, 46444, 45347,
+        44227, 43084, 41920, 40738, 39540, 38328, 37104, 35872, 34633, 33390, 32146, 30903, 29664,
+        28432, 27208, 25996, 24798, 23616, 22452, 21309, 20189, 19092, 18023, 16981, 15968, 14987,
+        14037, 13120, 12237, 11389, 10576, 9799,  9057,  8351,  7681,  7047,  6448,  5884,  5354,
+        4858,  4395,  3963,  3562,  3191,  2849,  2534,  2246,  1982,  1742,  1524,  1328,  1151,
+        993,   852,   726,   615,   518,   433,   359,   295,   240,   194,   155,   122,   94,
+        72,    54,    39,    28,    19,    12,    7,     4,     1};
```

Comment:
These constexpr arrays don't need to be in a header file. They are just used in one place.